lib/property_sets/active_record_extension.rb

```diff
@@ -34,41 +39,60 @@ def property_set(association, options = {}, &block)
           :inverse_of => self.name.underscore.to_sym,
         }.merge(options)
 
-        silence_warnings do
+        # TODO: should check options are compatible? warn? raise?
+        reflection = self.reflections[association.to_s] # => ActiveRecord::Reflection::HasManyReflection
+        reflection.options.merge! options if reflection && !options.empty?
+
+        unless exists then # makes has_many idempotent...
           has_many association, hash_opts do
-            include PropertySets::ActiveRecordExtension::AssociationExtensions
-
-            property_class.keys.each do |key|
-              raise "Invalid property key #{key}" if self.respond_to?(key)
-
-              # Reports the coerced truth value of the property
-              define_method "#{key}?" do
-                type  = property_class.type(key)
-                value = lookup_value(type, key)
-                ![ "false", "0", "", "off", "n" ].member?(value.to_s.downcase)
-              end
-
-              # Returns the value of the property
-              define_method "#{key}" do
-                type = property_class.type(key)
-                lookup_value(type, key)
-              end
-
-              # Assigns a new value to the property
-              define_method "#{key}=" do |value|
-                instance = lookup(key)
-                instance.value = PropertySets::Casting.write(property_class.type(key), value)
-                instance.value
-              end
-
-              define_method "#{key}_record" do
-                lookup(key)
-              end
+            # keep this damn block! -- creates association_module below
+          end
+        end
+
+        # eg 5: AccountSettingsAssociationExtension
+        # eg 6: Account::SettingsAssociationExtension
```

Comment:
I don't completely understand this. Where in Rails 5, `association.to_s.camelize` resolves to `AccountSettings`, in Rails 6 it resolves to `Settings` and `name.demodulize` resolves to `Account::`? Or is `association.to_s.camelize` empty and `name.demodulize` is `Account::Settings`?

Reply:
The code in AR is literally:

```ruby
"#{name.demodulize}#{association.to_s.camelize}AssociationExtension" # 4: "AccountSettingsAssociationExtension"
"#{association.to_s.camelize}AssociationExtension" # 5: "SettingsAssociationExtension"
```

but then the target of the `const_set` is also different:

```ruby
target = self.parent # 4: Object
target = self # 5: Account
```